ext/node/polyfills/net.ts

```diff
@@ -796,6 +801,19 @@ export class Socket extends Duplex {
 
     super(options);
 
+    // Note: If the socket is created from @npmcli/agent, the 'socket' event
+    // on ClientRequest object happens after 'connect' event on Socket object.
+    // That swaps the sequence of op_node_http_request_with_conn() call and
+    // initial socket read. That causes op_node_http_request_with_conn() not
+    // working.
+    // To avoid the above situation, we detect the socket created from
+    // @npmcli/agent and pause the socket (and also skips the startTls call
+    // if it's TLSSocket)
+    this._isNpmAgent = new Error().stack?.includes("@npmcli/agent") || false;
```

Comment:
Oh my, can you give more context on this problem? Do we know why `socket` happens after `connect`? Is it something the library does or is it the limitation of our implementation?

Reply:
agent can reuse a socket from its pool which is already connected

this is a hack we need to find a better solution later

cc @kt3k

Reply:
> Is it something the library does or is it the limitation of our implementation?

It is how `@npmcli/agent` works. The same thing happens in Node.js, and it's fine in Node.js.

This is only issue for us because our implementation uses the connection in conflicting way in `net` and `http`. In this branch in `http` we want hyper to take over the control of connection. So that op needs to happen before the first read of connection in `net` (which happens at `connect` event).

Reply:
Are we sure this is only the case for this `@npmcli/agent` library? I'm a bit worried that we'll need to expand this list in the future

Reply:
I think we are not sure at this point, but this is the only one we met within our test cases via `node-gyp` spec test (which depends on `@npmcli/agent` internally).

Other note is that there's more popular agent `npm:agent-base` and the basic usage of that agent didn't cause this issue. So I guess this issue affects relatively minor part of ecosystem.

Reply:
Okay, that seems reasonable